src/sallyport/mod.rs

```diff
@@ -566,4 +562,42 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn copy_into_slice() -> std::result::Result<(), OutOfSpace> {
+        let mut block = Block::default();
+
+        let c = block.cursor();
+        let (c, slab1) = c
+            .copy_from_slice::<usize>(&[1, 2])
+            .expect("allocate slab of 2 usize values for the first time");
+
+        let (c, slab2) = c
+            .copy_from_slice::<usize>(&[3, 4])
+            .expect("allocate slab of 2 usize values for the second time");
+
+        let (_c, slab3) = c
+            .copy_from_slice::<usize>(&[5, 6])
+            .expect("allocate slab of 2 usize values for the third time");
+
+        assert_eq!(slab1, &[1, 2]);
+        assert_eq!(slab2, &[3, 4]);
+        assert_eq!(slab3, &[5, 6]);
+
+        let c = block.cursor();
+
+        let mut slab_all = [0usize; 3];
+
+        let c = unsafe { c.copy_into_slice::<usize>(4, &mut slab_all, 3) }?;
```

Comment:
Why is the `src_len` a `4` here? That is shorter than the total slab 1-3 values but longer than `slab_all`.

Reply:
That is the amount of elements the cursor should advance after copying.